apps/src/templates/instructions/TopInstructionsCSF.jsx

```diff
@@ -134,7 +173,39 @@ var TopInstructions = React.createClass({
   },
 
   getInitialState() {
-    return { rightColWidth: 90 };
+    return {
+      rightColWidth: 90,
+      promptForHint: false
+    };
+  },
+
+  componentDidUpdate(prevProps) {
+    // Update right col width now that we know how much space it needs, and
+    // rerender if it has changed. One thing to note is that if we end up
+    // resizing our column significantly, it can result in our maxNeededHeight
+    // being inaccurate. This isn't that big a deal except that it means when we
+    // adjust maxNeededHeight below, it might not be as large as we want.
+    var width = $(ReactDOM.findDOMNode(this.refs.collapser)).outerWidth(true);
+    if (width !== this.state.rightColWidth) {
+      this.setState({
+        rightColWidth: width
+      });
+    }
+
+    const gotNewFeedback = this.props.feedback && !prevProps.feedback;
+    if (gotNewFeedback) {
+      this.setState({
+        promptForHint: false
+      });
```

Comment:
I personally like to do setState with a single item on one line (though how you have it is fine). Could change this and a few other places if you like, i.e.
```
this.setState({ promptForHint: false })